### PR TITLE
[CC-20333] CVE fix for jackson package

### PR DESCRIPTION
**Pull request by sp-gupta**
## Problem
https://confluentinc.atlassian.net/browse/CC-20333

## Solution
Fixed to 2.15.0 version 
Screenshot 2023-07-31 at 6 43 05 PM

##### Does this solution apply anywhere else?
- [ ] yes
- [x] no

##### Testing done:
- [x] Unit tests
- [x] Integration tests
- [ ] System tests
- [x] Manual tests
              - Docker Playground
          
```
Creating Elasticsearch Sink connector (Elasticsearch version is 7.0.0)
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   634  100   334  100   300   5220   4688 --:--:-- --:--:-- --:--:-- 10745
{
  "name": "elasticsearch-sink",
  "config": {
    "connector.class": "io.confluent.connect.elasticsearch.ElasticsearchSinkConnector",
    "tasks.max": "1",
    "topics": "test-elasticsearch-sink",
    "key.ignore": "true",
    "connection.url": "http://elasticsearch:9200",
    "name": "elasticsearch-sink"
  },
  "tasks": [
    {
      "connector": "elasticsearch-sink",
      "task": 0
    }
  ],
  "type": "sink"
}
Sending messages to topic test-elasticsearch-sink
11:16:31 ℹ️ 🔮 schema was identified as avro
11:16:31 ℹ️ ✨ generating data...
11:16:31 ℹ️ ☢️ --forced-value is set
11:16:31 ℹ️ ✨ 10 records were generated based on --forced-value  (only showing first 10), took: 0min 1sec
{"f1":"value1"}
{"f1":"value2"}
{"f1":"value3"}
{"f1":"value4"}
{"f1":"value5"}
{"f1":"value6"}
{"f1":"value7"}
{"f1":"value8"}
{"f1":"value9"}
{"f1":"value10"}
11:16:35 ℹ️ 💯 Get number of records in topic test-elasticsearch-sink
0
11:16:35 ℹ️ 🛡️ Set compatibility for subject test-elasticsearch-sink-value to NONE
{"compatibility":"NONE"}11:16:35 ℹ️ 📤 producing 10 records to topic test-elasticsearch-sink
11:16:37 ℹ️ 📤 produced 10 records to topic test-elasticsearch-sink, took: 0min 2sec
11:16:37 ℹ️ 💯 Get number of records in topic test-elasticsearch-sink
10
Check that the data is available in Elasticsearch
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  2446  100  2446    0     0  32522      0 --:--:-- --:--:-- --:--:-- 34942
{
  "took" : 54,
  "timed_out" : false,
  "_shards" : {
    "total" : 1,
    "successful" : 1,
    "skipped" : 0,
    "failed" : 0
  },
  "hits" : {
    "total" : {
      "value" : 10,
      "relation" : "eq"
    },
    "max_score" : 1.0,
    "hits" : [
      {
        "_index" : "test-elasticsearch-sink",
        "_type" : "_doc",
        "_id" : "test-elasticsearch-sink+0+0",
        "_score" : 1.0,
        "_source" : {
          "f1" : "value1"
        }
      },
      {
        "_index" : "test-elasticsearch-sink",
        "_type" : "_doc",
        "_id" : "test-elasticsearch-sink+0+1",
        "_score" : 1.0,
        "_source" : {
          "f1" : "value2"
        }
      },
      {
        "_index" : "test-elasticsearch-sink",
        "_type" : "_doc",
        "_id" : "test-elasticsearch-sink+0+2",
        "_score" : 1.0,
        "_source" : {
          "f1" : "value3"
        }
      },
      {
        "_index" : "test-elasticsearch-sink",
        "_type" : "_doc",
        "_id" : "test-elasticsearch-sink+0+3",
        "_score" : 1.0,
        "_source" : {
          "f1" : "value4"
        }
      },
      {
        "_index" : "test-elasticsearch-sink",
        "_type" : "_doc",
        "_id" : "test-elasticsearch-sink+0+4",
        "_score" : 1.0,
        "_source" : {
          "f1" : "value5"
        }
      },
      {
        "_index" : "test-elasticsearch-sink",
        "_type" : "_doc",
        "_id" : "test-elasticsearch-sink+0+5",
        "_score" : 1.0,
        "_source" : {
          "f1" : "value6"
        }
      },
      {
        "_index" : "test-elasticsearch-sink",
        "_type" : "_doc",
        "_id" : "test-elasticsearch-sink+0+6",
        "_score" : 1.0,
        "_source" : {
          "f1" : "value7"
        }
      },
      {
        "_index" : "test-elasticsearch-sink",
        "_type" : "_doc",
        "_id" : "test-elasticsearch-sink+0+7",
        "_score" : 1.0,
        "_source" : {
          "f1" : "value8"
        }
      },
      {
        "_index" : "test-elasticsearch-sink",
        "_type" : "_doc",
        "_id" : "test-elasticsearch-sink+0+8",
        "_score" : 1.0,
        "_source" : {
          "f1" : "value9"
        }
      },
      {
        "_index" : "test-elasticsearch-sink",
        "_type" : "_doc",
        "_id" : "test-elasticsearch-sink+0+9",
        "_score" : 1.0,
        "_source" : {
          "f1" : "value10"
        }
      }
    ]
  }
}
          "f1" : "value1"
          "f1" : "value10"
          "f1" : "value10"
          
```

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -49,7 +49,7 @@
         <apacheds-jdbm1.version>2.0.0-M2</apacheds-jdbm1.version>
         <!-- TODO: Remove the version pin after releasing https://github.com/confluentinc/common/pull/494 -->
         <jackson.databind.version>2.14.0</jackson.databind.version>
```
[the remark this answers is not in the document]
Comment (sp-gupta) [reply pom.xml:51]: Sure, we should do it

File: Jenkinsfile [hunk 2]
```diff
@@ -4,4 +4,5 @@ common {
   upstreamProjects = 'confluentinc/common'
   pintMerge = true
   downStreamValidate = false
+  nodeLabel = 'docker-debian-jdk8'
```
Comment (sp-gupta) [inline Jenkinsfile:7]: This change is to prevent builds from failing.

Comment (subhashiyer9) [reply Jenkinsfile:7]: Any reference on why we needed to add this now?
Anything changed recently?

Comment (sp-gupta) [reply Jenkinsfile:7]: @subhashiyer9 [This](https://github.com/confluentinc/kafka-overlay/pull/176#issue-1824790544) description somewhat explains the change

**Review by subhashiyer9 — APPROVED** [review]

**Pull request merged**